### DISCUSSION TITLE
Properly formatting a query parameter for deepObject style

### DISCUSSION THREAD

**Discussion by rgates-fieldwire** [rswag/rswag#526] [General]
I've found a couple of similar questions around this topic but none of them seem to answer my questions. 

Our API has a filter option that can be passed in as a query parameter:

Example: HOST/api/v3/projects?filters[name_like]=Test&filters[created_at_gte]=2022-04-02 03:55:27.748Z

Looking over [Swagger docs](https://swagger.io/docs/specification/describing-parameters/), Style: deepObject, Explode: True. Should be the only style that I need 

I define the parameter:
`parameter name: 'filters', in: :query, schema: {
        type: :object
      },
      style: :deepObject,
      explode: true`

and then define it:

```
let('filters') do
            {
              'name_like' => 'test'
            }
          end
```

However I'm getting the following error:

```
URI::InvalidURIError:
       bad URI(is not URI?): http://www.example.com/api/v3/projects?filters={%22name_like%22=%3E%22test%22}
```

I have set up parameter in the style of  name:filters[#{name}] as work around but that is producing a very poor swagger doc on generation.

We are using gem version: rswag-api (2.3.1), rswag-specs (2.4.0)

**Comment by flarecriteria** [comment]
Way late, but does it work with the properties specified?

```ruby
      parameter name: :filters, in: :query, schema: {
        type: :object,
        style: :deepObject,
        explode: true,
        properties: {
          name_like: { type: :string },
          created_at_gte: { type: :string }
        }
      }
```

**Comment by jtannas** [comment]
Moving to https://github.com/rswag/rswag/issues/582
